common/persistence/task_manager.go

```diff
@@ -233,6 +233,14 @@ func (m *taskManagerImpl) CreateTasks(request *CreateTasksRequest) (*CreateTasks
 }
 
 func (m *taskManagerImpl) GetTasks(request *GetTasksRequest) (*GetTasksResponse, error) {
+	if request.MinTaskID >= request.MaxTaskID {
```

Comment:
the == case might be legit, or not? If there is no task written since last read, but we trigger a new read, would the min == max? Or should the upper layer handle that case?

Reply:
min task ID is exclusive
max task ID is inclusive

i can relax the check for now